### PR TITLE
JENKINS-38012# Fix related to Kyoto related failure handling

### PR DESCRIPTION
**Pull request by vivek**
# Description

See https://issues.jenkins-ci.org/browse/JENKINS-38012?filter=-1

* Run a pipeline project https://github.com/vivek/hellokyoto/blob/mic/Jenkinsfile
* It should show 'Build' stage as failed in blueocean UI

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

Ran locally, works fine.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by vivek** [comment]
CI is failing due to kyoto snapshot dependency. will be fixed once 0.2 is released.

**Comment by i386** [comment]
ping @abayer

**Comment by i386** [comment]
@vivek tested locally - LGTM 🐝

**Comment by vivek** [comment]
@i386 ok, will up the version after 0.2 is released and one more round of testing then will merge:)

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeUtil.java [hunk 1]
```diff
@@ -47,8 +47,20 @@
...
+                    result = BlueRun.BlueRunResult.SUCCESS;
+                    break;
+                case "ABORTED":
+                    result = BlueRun.BlueRunResult.ABORTED;
+                    break;
+                case "FAILURE":
+                    result = BlueRun.BlueRunResult.FAILURE;
+                    break;
+                default:
+                    result = BlueRun.BlueRunResult.UNKNOWN;
```
Comment (tfennelly) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeUtil.java]: Seems like the above `switch` could be shortened to ... 

```java
try {
    result = BlueRun.BlueRunResult.valueOf(run.getResult().toString());
} catch (RuntimeException e) {
    result = BlueRun.BlueRunResult.UNKNOWN;
}
```

That also covers states that might be added in the future.

**Comment by tfennelly** [comment]
@vivek LGTM so far. Great job on the tests (as usual !!!).

**Comment by abayer** [comment]
:bee:

**Pull request merged**